### ISSUE TITLE
ConnectionException Login Bug

### ISSUE THREAD

**Issue by Physsikal** [instaloader/instaloader#1954]
### Bug Description
When using the `login()` function, a `ConnectionException` gets raised that redirects to a Instagram URL. However, when going through to the given URL the user is met with the following screen:

![instaloader Login Error](https://user-images.githubusercontent.com/77701072/233850599-0bc551c0-51d7-41f8-bab4-702ad0be7b7d.png)

Every time when the user goes trough to the link that is being given in the exception he gets redirected to the _default_ Instagram login page (not a 2FA page). After logging in the password reset screen is prompted and all other session are logged out and the account becomes unusable until the password is changed.

### Steps To Reproduce

The code used to produce the bug is:
```python
import instaloader

L = instaloader.Instaloader()

L.login('*******', '*********')

profile = instaloader.Profile.from_username(L.context, '************')
``` 

### Expected Behaviour

The expected result would be that after clicking the link in the error message of the terminal I get redirected to a 2FA page of Instagram where I confirm my identity, and after retrying the script I am not prompted the 2FA anymore and can log in.

### Error Messages and Tracebacks

```shell
PS C:\Users\Redacted> & "C:/Program Files/Python311/python.exe" c:/Users/Redacted/Desktop/InstaTRACKER/testingPage.py
Traceback (most recent call last):
  File "c:\Users\Redacted\Desktop\InstaTRACKER\testingPage.py", line 5, in <module>
    L.login('*******', '*********')
  File "C:\Users\Redacted\AppData\Roaming\Python\Python311\site-packages\instaloader\instaloader.py", line 634, in login
    self.context.login(user, passwd)
  File "C:\Users\Redacted\AppData\Roaming\Python\Python311\site-packages\instaloader\instaloadercontext.py", line 239, in login
    raise ConnectionException("Login: Checkpoint required. Point your browser to "
instaloader.exceptions.ConnectionException: Login: Checkpoint required. Point your browser to https://www.instagram.com/challenge/action/REDACTED/ - follow the instructions, then retry.
``` 

### Instaloader Version

`instaloader 4.9.6`

### Additional context

I am aware about the current workaround [by importing the session cookies from Firefox](https://raw.githubusercontent.com/instaloader/instaloader/master/docs/codesnippets/615_import_firefox_session.py). Nonetheless, I want to create a script that does not require the user to have Firefox installed.

**Comment by Showmaster3000** [comment]
see pinned #1758 .

**Comment by Physsikal** [comment]
> see pinned #1758 .

In the last part of my reply I already mentioned that I tried the cookie export method from Firefox. It works for me but it's not the most elegant solution and requires the user to _have to_ install Firefox.

I have tried to create a automation script using `selenium` but the problem is that the cookie files do not get saved to the local Firefox profile and therefore cannot be extracted using `sqlite3`.

The workaround is good enough if you're using it for yourself but I can't seem to find a method in order to be able to use the  `instaloader` package for a general distribution case.
